packages/core/src/tools/codegen/extract-files.ts

```diff
@@ -54,31 +63,74 @@ function _extractFiles(args: ExtractFilesConfig) {
       debug.common('Skipping code extraction, code generation has been disabled');
       return;
     }
+
     console.log(chalk.green('Code extraction started'));
     const basePath = process.cwd();
 
     try {
-      // Use Edge Platform mesh endpoint - staging is ready, prod QA in progress
       const targetUrl = args.scConfig.api.edge.edgeUrl;
       const { accessToken } = await auth.clientCredentialsFlow(authParams);
       if (!accessToken) {
         console.error(chalk.red('Failed to get access token, aborting code extraction'));
         return;
       }
 
-      const componentPaths = await resolveComponentImportFiles(basePath, args.componentMapPath);
+      // Resolve files from component-map
+      const resolvedImports = await resolveComponentImportFiles(basePath, args.componentMapPath);
 
-      const fileDispatches = Array.from(componentPaths, (mapEntry) =>
-        sendCode({
-          file: {
-            name: mapEntry[0],
-            path: mapEntry[1],
-            type: ExtractedFileType.Component,
-          },
-          token: accessToken,
-          targetUrl,
-        })
-      );
+      let items: Array<{ componentKey: string; filePath: string }> = [];
+
+      if (resolvedImports && typeof (resolvedImports as ResolveResult).imports !== 'undefined') {
```

Comment:
this should ideally be done under `resolveComponentImportFiles`. I also feel extraLabels overcomplicates the logic, it's very hard to follow what is being done here (but `resolveComponentImportFiles` refactor is very welcome)

Reply:
i have refactored and simplified the extract-files function a bit so now it should seem simple to understand.